winch/codegen/src/isa/x64/masm.rs

```diff
@@ -1335,11 +1335,25 @@ impl Masm for MacroAssembler {
                 self.asm
                     .lock_xadd(addr, operand.to_reg(), operand, size, flags);
             }
+            RmwOp::And | RmwOp::Or | RmwOp::Xor => {
+                let op = match op {
+                    RmwOp::And => AtomicRmwSeqOp::And,
+                    RmwOp::Or => AtomicRmwSeqOp::Or,
+                    RmwOp::Xor => AtomicRmwSeqOp::Xor,
+                    _ => unreachable!(
+                        "invalid op for atomic_rmw_seq, should be one of `or`, `and` or `xor`"
+                    ),
```

Comment:
Could you return an error here instead of panicking?

Reply:
does it make sense to return an error here? we explicitely check that we only match the right variants above, this branch should be legitmately unreachable. No `CodeGenError` seem to correspond, so I was reluctant to add an error variant for a trivially unreachable statement.

Reply:
Yeah you're right, it's probably fine as is. I'm trying to error on the side of caution when it comes to panics, because we've put a lot of effort to make sure that they are minimal, particularly if they affect partial development of features at the visitor/masm layer. Here I got confused by the nested match which led me to think that we were missing an implementation.